deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,244 @@
+"""Module containing robustness report check."""
+from copy import copy
+
+import numpy as np
+import random
+import torch
+import imgaug
+from typing import Callable, TypeVar, List, Optional
+import albumentations
+
+import pandas as pd
+import plotly.express as px
+from ignite.metrics import Metric
+
+from deepchecks import CheckResult
+from deepchecks.vision import VisionDataset, SingleDatasetCheck, Context
+from deepchecks.vision.dataset import TaskType
+from deepchecks.vision.metrics_utils import calculate_metrics
+
+__all__ = ['RobustnessReport']
+
+from vision.utils.transformations import add_augmentation
+
+PR = TypeVar('PR', bound='RobustnessReport')
+
+
+class RobustnessReport(SingleDatasetCheck):
+    """Check several image enhancements for model robustness.
+
+    Parameters
+    ----------
+        alternative_metrics (List[Metric], default None):
+            A list of ignite.Metric objects whose score should be used. If None are given, use the default metrics.
+    """
+
+    def __init__(self,
+                 alternative_metrics: Optional[List[Metric]] = None,
+                 prediction_extract: Optional[Callable] = None,
+                 augmentations=None,
+                 epsilon: float = 10 ** -2,
+                 random_seed: int = 42):
+        super().__init__()
+        self._epsilon = epsilon
+        self.alternative_metrics = alternative_metrics
+        self.prediction_extract = prediction_extract
+        self.random_seed = random_seed
+        self.augmentations = augmentations
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
```

Comment:
This should run on test by default.

Reply:
This doesn't work like that. this is way SingleDatasetCheck behave

Reply:
so what's the role of dataset_type: str = 'train' ?

Reply:
I mean it is the role, but it's fixed for SingleDatasetCheck, it's not per-check configuration